src/Hl7.Fhir.Core/Rest/SearchParams.cs

```diff
@@ -177,6 +183,20 @@ private static void addNonEmpty(string paramName, IList<string> values, string v
             values.Add(value);
         }
 
+        private static void addNonEmptyInclude(string paramName, IList<Tuple<string,List<string>>> values, string value)
+        {
+            if (String.IsNullOrEmpty(value)) throw Error.Format("Invalid {0} value: it cannot be empty".FormatWith(paramName));
+            if ( paramName == SEARCH_PARAM_INCLUDE)
+            {
+                values.Add(new Tuple<string, List<string>>(value, new List<string>()));
+            }
+            else // ITERATIVE INCLUDE
+            {
+                if ( !values.Any() ) throw Error.Format("Invalid {0}: must be preceded by a regular include".FormatWith(paramName));
```

Comment:
We don't support circular relationships in any of our concepts? I guess we should still support it because this technically isn't CareEvolution specific.

Reply:
Yes, I would try to keep this as generic as possible